runtime/lua/vim/lsp/util.lua

```diff
@@ -1290,18 +1290,159 @@ function M._trim(contents, opts)
   return contents
 end
 
---- Generates a table mapping markdown code block lang to vim syntax,
---- based on g:markdown_fenced_languages
----@return table table of lang -> syntax mappings
-local function get_markdown_fences()
-  local fences = {}
-  for _, fence in pairs(vim.g.markdown_fenced_languages or {}) do
-    local lang, syntax = fence:match('^(.*)=(.*)$')
-    if lang then
-      fences[lang] = syntax
-    end
-  end
-  return fences
+---@alias MarkdownBlock {line:string}
+---@alias MarkdownCodeBlock {code:string[], lang:string}
+---@alias Markdown (MarkdownBlock|MarkdownCodeBlock)[]
+
+---@param text string
+---@param opts? {ft?: string}
+---@return Markdown
+local function parse_markdown(text, opts)
+  opts = opts or {}
+  text = text:gsub('</?pre>', '```'):gsub('\r', '')
+
+  -- Replace common HTML entities.
+  local entities =
+    { nbsp = '', lt = '<', gt = '>', amp = '&', quot = '"', apos = "'", ensp = ' ', emsp = ' ' }
+  for entity, char in pairs(entities) do
+    text = text:gsub('&' .. entity .. ';', char)
+  end
```

Comment:
You can use a table with `string.gsub`:

```suggestion
  local entities =
    { nbsp = '', lt = '<', gt = '>', amp = '&', quot = '"', apos = "'", ensp = ' ', emsp = ' ' }
  text = test:gsub('&([^ ;]+);', entities)
```

This actually exists on master today: https://github.com/neovim/neovim/blob/1f551e068f728ff38bd7fdcfa3a6daf362bab9da/runtime/lua/vim/lsp/util.lua#L1398-L1410

Reply:
Oops, this was one of the bits I accidentally replaced when porting the bits from `noice`...